classes/observers.php

```diff
@@ -556,4 +553,48 @@ public static function course_module_updated(\core\event\course_module_updated $
             }
         }
     }
+
+    /**
+     * Event handler for the user enrolment updated event. It unenrols/enrols the user
+     * from mattermost channels when the user is suspended or made active in the course
+     *
+     * @param \core\event\user_enrolment_updated $event
+     */
+    public static function user_enrolment_updated(\core\event\user_enrolment_updated $event) {
+        global $DB;
+        $userenrolmentid = $event->objectid;
+        $userid = $event->relateduserid;
+        $courseid = $event->courseid;
+        $userenrolment = $DB->get_record('user_enrolments', array('id' => $userenrolmentid));
+        if (!$userenrolment) {
+            throw new moodle_exception('userenrolmentnotfounderror', 'mod_mattermost');
+        }
+
+        $mattermostinstance = $DB->get_record('mattermost', array('course' => $courseid));
+        if (!$mattermostinstance) {
+            return;
+        }
+
+        $channelids = array($mattermostinstance->mattermostid);
+        $groups = $DB->get_records('mattermostxgroups', array('courseid' => $courseid));
+        foreach ($groups as $group) {
```

Comment:
should we also add check here to proceed if `$groups` exists, as it may give error if someone manually deleted this data from DB

Reply:
There's no need for checking this. The `get_records` function returns an empty array if no record exists.